src/service/extension-location.js

```diff
@@ -105,10 +126,23 @@ export function calculateEntryPointScriptUrl(
 export function parseExtensionUrl(scriptUrl) {
   // Note that the "(\.max)?" group only applies to local dev.
   const matches = scriptUrl.match(
-    /^(.*)\/(.*)-([0-9.]+|latest)(\.max)?\.(?:js|mjs)$/i
+    /^(.*)\/(.*)-([0-9.]+|latest)(\.max)?\.(?:js|mjs)(\?.*)?$/i
   );
   return {
     extensionId: matches ? matches[2] : undefined,
     extensionVersion: matches ? matches[3] : undefined,
   };
 }
+
+/**
+ * Add additional values to the script url depending on mode, such as query parameters.
+ * @param {string} scriptUrl
+ * @param {boolean=} opt_isLocalDev
+ * @return {string}
+ */
+function finalizeScriptUrl(scriptUrl, opt_isLocalDev) {
```

Comment:
can we rename this? doesn't seem to be descriptive

Reply:
Renamed this to `maybeAddQueryParams`.